### PR TITLE
Fixed #26097 -- Added password_validators_help_text_html to UserCreationForm for consistency with other django.contrib.auth forms.

### PR DESCRIPTION
**Pull request by sasha0**
Ticket - https://code.djangoproject.com/ticket/26097

### CODE REVIEW

**Comment by sasha0** [comment]
@timgraham do you think a test is necessary?

File: django/contrib/admin/static/admin/css/base.css [hunk 1]
```diff
@@ -187,9 +187,15 @@ p.mini {
...
 
-.help, p.help, form p.help {
+.help, span.help, form span.help, form span.help li {
     font-size: 11px;
     color: #999;
+    display: block;
+}
+
+form span.help ul {
+    margin-left: 0px !important;
```
Comment (timgraham) [inline django/contrib/admin/static/admin/css/base.css]: There's a ticket about removing `!important` from the admin's CSS, so we should try not to introduce more of that.
Also "0px" should be "0".

**Comment by timgraham** [comment]
I think a test shouldn't be too difficult.

I'm not sure about the  p -> span changes though. This might be a bit disruptive for anyone extending the admin with custom markup or CSS. I don't have a good solution in mind.

**Comment by sasha0** [comment]
> I'm not sure about the p -> span changes though. This might be a bit disruptive for anyone extending the admin with custom markup or CSS. I don't have a good solution in mind.

Otherwise, we could output `help_text` for password validators in another way, without `<ul>` tag. What do you think?

**Comment by sasha0** [comment]
> I think a test shouldn't be too difficult.

Something like this?

```python
@override_settings(AUTH_PASSWORD_VALIDATORS=ALL_AUTH_PASSWORD_VALIDATORS)
 def test_password1_help_text(self):
     form = UserCreationForm()
     self.assertEqual(form.fields['password1'].help_text, 
                      ALL_PASSWORD_VALIDATORS_HELP_TEXT_HTML)
```

**Comment by timgraham** [comment]
Yes about the test. I would ask for feedback on the mailing list about the HTML changes.

**Comment by sasha0** [comment]
Alright, I'll ask.

**Comment by sasha0** [comment]
Ok, asked about this in the [mailing list](https://groups.google.com/forum/#!topic/django-
 developers/nAk3zrMDeNc).

**Comment by sasha0** [comment]
Unfortunately, there's no feedback so far.

**Comment by berkerpeksag** [comment]
I'd use `<div>` instead of `<span>`, add a special CSS rule for `div.help` and leave other `<p class="help">` usages unmodified.

**Comment by sasha0** [comment]
@berkerpeksag yeah, nice idea, thanks.

@timgraham what do you think?

**Comment by timgraham** [comment]
Okay with me. Let's just mention it in the release notes.

**Comment by sasha0** [comment]
> Okay with me. Let's just mention it in the release notes.

File: django/contrib/admindocs/templates/admin_doc/bookmarklets.html [hunk 2]
```diff
@@ -12,11 +12,11 @@
 
 {% block content %}
 
-<p class="help">{% blocktrans trimmed %}
+<div class="help">{% blocktrans trimmed %}
```
Comment (timgraham) [inline django/contrib/admindocs/templates/admin_doc/bookmarklets.html]: No need to change this, right?

File: django/contrib/auth/forms.py [hunk 3]
```diff
@@ -72,6 +72,7 @@ class UserCreationForm(forms.ModelForm):
         label=_("Password"),
         strip=False,
         widget=forms.PasswordInput,
+        help_text=password_validation.password_validators_help_text_html()
```
Comment (timgraham) [inline django/contrib/auth/forms.py]: add trailing comma so if more items are later added, we don't have to modify this line again.

File: docs/releases/1.10.txt [hunk 4]
```diff
@@ -734,6 +734,9 @@ Miscellaneous
   model, you must convert them to attributes or properties as described in
   :ref:`the deprecation note <user-is-auth-anon-deprecation>`.
 
+* In the admin templates, <p class="help"> was replaced with div tag for the
```
Comment (timgraham) [inline docs/releases/1.10.txt]: Add double backticks around HTML stuff so it renders as "code".
"with a div tag to allow including lists inside help messages"

File: tests/test_utils/tests.py [hunk 5]
```diff
@@ -653,10 +653,10 @@ def test_contains_html(self):
             self.assertContains(response, '<p "whats" that>')
 
     def test_unicode_handling(self):
-        response = HttpResponse('<p class="help">Some help text for the title (with unicode ŠĐĆŽćžšđ)</p>')
+        response = HttpResponse('<span class="help">Some help text for the title (with unicode ŠĐĆŽćžšđ)</span>')
```
Comment (timgraham) [inline tests/test_utils/tests.py]: no need to change this file

**Comment by timgraham** [comment]
There can be a first commit with the `<p>` to `<div>` changes, then another commit for actually adding the `password_validators_help_text_html`.

**Comment by sasha0** [comment]
Alright, I'll split changes into 2 commits.

**Comment by timgraham** [comment]
How about a test for the second commit as discussed in the earlier comments?

**Comment by sasha0** [comment]
Of course, will add.

**Comment by timgraham** [comment]
Have time to finish this up? I think we should have a ticket for that first commit since it's not exactly clear what it fixes (just to have some reference in case it causes some regression).

**Comment by sasha0** [comment]
Yep, will finish it soon.

**Comment by timgraham** [comment]
merged in 2d26b4637f0318736955b54b34f270fe9500f795 and 536db42cf01ebe0dd633b131c3dbfc0bf743064f, thanks!

Please open a new ticket for that first commit if it's still needed.

**Pull request closed without merging**